CONTRIBUTING.rst

```diff
@@ -0,0 +1,1258 @@
+|TITLE| (|DATE|)
+##################################
+
+.. |TITLE| replace:: OCA Guidelines
+.. |DATE| replace:: 2018/08/24
+.. |RELATED| replace:: OCA Guidelines
+.. |CATEGORIES| replace:: Guidelines, Conventions
+
+.. fill in the Document information in the above lines. It will be automatically
+   replaced in the document header. Don't forget to remove the []
+
+.. contents:: Table of contents
+    :depth: 4
+
+.. sectnum::
+
+
+
+************
+Introduction
+************
+
+This page introduces the coding guidelines for projects hosted under OCA. These
+guidelines aim to improve the quality of the code: better readability of
+source, better maintainability, better stability and fewer regressions.
+
+These are loosely based on the `Odoo Guidelines <https://www.odoo.com/documentation/8.0/reference/guidelines.html>`_
+and `Old Odoo Guidelines <https://doc.odoo.com/contribute/15_guidelines/coding_guidelines_framework.html>`_
+with adaptations to improve their guidelines and make them more suitable for
+this project's own needs. Readers used to the Odoo Guidelines can skip to the
+`Differences With Odoo Guidelines <#id1>`_ section.
+
+*******
+Modules
+*******
+
+* Use of the singular form in module name (or use "multi"),
+  except when compound of module name or object Odoo
+  that is already in the plural (i.e. `mrp_operations_...`).
+* If your module's purpose is to serve as a base for other modules, prefix its
+  name with `base_`. I.e. `base_location_nuts`.
+* When creating a localization module, prefix its name with `l10n_CC_`, where
+  `CC` is its country code. I.e. `l10n_es_pos`.
+* When extending an Odoo module, prefix yours with that module's name. I.e.
+  `mail_forward`.
+* When combining an Odoo module with another from OCA, Odoo's name goes before.
+  I.e., if you want to combine Odoo's `crm` with OCA's `partner_firstname`, the
+  name should be `crm_partner_firstname`.
+* Use the `description template <https://github.com/OCA/maintainer-tools/tree/master/template/module>`_
+  but remove sections with no meaningful content.
+* In the `__openerp__.py`/`__manifest__.py`  manifest file:
+
+  * Avoid empty keys
+  * Make sure it has the `license` and `images` keys.
+  * Make sure the text `,Odoo Community Association (OCA)` is appended to the
+    `author` text.
+  * The `website` key must be `https://github.com/OCA/<repo>`,
+    so as to provide the most relevant link to discover more information about the addon.
+    That link shows the repository README. Alternatively `https://github.com/OCA/<repo>/tree/<branch>/<addon>`
+    may be used, to provide a direct link to the addon README, which includes proper credits
+    (authors, contributors and theirs companies), and links to the relevant information on the OCA website.
+* Don't use your company logo or your corporate branding. Using the author and the list of contributors is enough for people to know about your employer/company and contact you.
+
+***************
```

Comment:
I think this level is not correct: should be underlined with ========

Reply:
@elicoidal There is 4 levels in this document. I tried various characters to structure it correctly. If you look at the output, goal is achieved.

Reply:
@max3903 what I am trying to say is that this level is awkward (probably even in original document): there are 4 levels but 2 entries (Modules and Version Numbers) as top level makes no real sense (`Version number` is upper level of ` Migration` which seems not logical to me)
Here is the current structure:
![screenshot from 2018-09-05 08-40-38](https://user-images.githubusercontent.com/7600613/45064720-e9212e80-b0e7-11e8-8f53-2ee3a5ceaa90.png)

I suggest to change it to the following structure:
![screenshot from 2018-09-05 08-44-00](https://user-images.githubusercontent.com/7600613/45064726-f0e0d300-b0e7-11e8-9fdf-1f332b0f0e27.png)